src/runtime/ranger/ranger_resource_policy.h

```diff
@@ -0,0 +1,102 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <set>
+#include <vector>
+
+#include <rapidjson/document.h>
+
+#include "common/json_helper.h"
+#include "utils/fmt_logging.h"
+
+namespace dsn {
+namespace ranger {
+
+// ACL type defined in Range service for RPC matching policy
+enum access_type
+{
+    INVALID = 0,
+    READ = 1,
+    WRITE = 1 << 1,
+    CREATE = 1 << 2,
+    DROP = 1 << 3,
+    LIST = 1 << 4,
+    METADATA = 1 << 5,
+    CONTROL = 1 << 6,
+    ALL = 1 << 7,
```

Comment:
When will `INVALID` and `ALL` be used? Are they necessary? Is it correct to set all bits for `ALL` but not only 1 bit?